chromadb/test/ef/test_ef_exceptions.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import os
+import random
+import string
+from typing import Generator, Optional
+
+import pytest
+from _pytest.logging import LogCaptureFixture
+from dotenv import load_dotenv
+
+from chromadb import EmbeddingFunction
+from chromadb.utils.embedding_functions import (
+    SentenceTransformerEmbeddingFunction,
+    DefaultEmbeddingFunction,
+    OpenAIEmbeddingFunction,
+    HuggingFaceEmbeddingFunction,
+    InstructorEmbeddingFunction,
+)
+
+
+def generate_random_sentence(word_count: int = 1025) -> str:
+    words = [
+        "".join(random.choices(string.ascii_lowercase, k=random.randint(3, 10)))
+        for _ in range(word_count)
+    ]
+    sentence = " ".join(words) + "."
+    return sentence
+
+
+def sentence_transformers_ef() -> Optional[EmbeddingFunction]:
+    return SentenceTransformerEmbeddingFunction(model_name="all-mpnet-base-v2")
+
+
+def instructor_ef() -> Optional[EmbeddingFunction]:
+    return InstructorEmbeddingFunction()
+
+
+def default_ef() -> Optional[EmbeddingFunction]:
+    return DefaultEmbeddingFunction()
+
+
+def openai_ef() -> Optional[EmbeddingFunction]:
+    load_dotenv()
+    api_key = os.getenv("OPENAI_API_KEY")
+    if api_key is None:
+        return None
+    return OpenAIEmbeddingFunction(api_key=api_key)
+
+
+def huggingface_ef() -> Optional[EmbeddingFunction]:
+    load_dotenv()
+    api_key = os.getenv("HUGGINGFACE_API_KEY")
+    if api_key is None:
+        return None
+    return HuggingFaceEmbeddingFunction(
+        model_name="sentence-transformers/all-MiniLM-L6-v2",
+        api_key=api_key,
+    )
+
+
+@pytest.fixture(
+    scope="function",
+    params=[
+        sentence_transformers_ef,
+        default_ef,
+        huggingface_ef,
+        instructor_ef,
```

Comment:
openai_ef is not here?

Reply:
OpenAI actually have validation input on their embedding endpoint, which will throw if the user exceeds model's max input tokens. So we just delegate this to the API.

Reply:
Let's make a comment to that effect.